eokwukwe/ecommerce-shop-api
# #169110656 User Registration

#### What does this PR do?
Implement user registration
 #### Description of Task to be completed?
- [x] Implement the controller functionality for a new user to register
 #### How should this be manually tested?
- clone the branch
- Run `npm install` to install dependencies
- Run `npm run dev` to start the development server
- Send a `POST` request to `http://localhost:80/api/v1/customers` with the following credentials:
```
{
  "name": "<fullname>",
  "email": "<email>",
  "password": "<password>"
}
```
- Expected output if the credentials are correct: 
```
{
  "customer": {
    "schema": {
      "customer_id": <id>,
      "name": "<name>",
      "email": "<email>",
      "address_1": null,
      "address_2": null,
      "city": null,
      "region": null,
      "postal_code": null,
      "country": null,
      "shipping_region_id": 1,
      "day_phone": null,
      "eve_phone": null,
      "mob_phone": null,
      "credit_card": null
    }
  },
  "accessToken": "Bearer <token.....>",
  "expires_in": "24h"
}
```
 #### Any background context you want to provide?
N/A
 #### What are the relevant pivotal tracker stories?
[#169110656](https://www.pivotaltracker.com/story/show/169110656)
